neo4j-jdbc-bolt/src/main/java/org/neo4j/jdbc/bolt/BoltNeo4jDatabaseMetaData.java

```diff
@@ -131,20 +131,18 @@ private void cycleResultSetToSetDatabaseProperties(Result rs, Table databaseLabe
 	 * @return
 	 */
 	private List<String> callDbmsFunctions(Session session){
-		List<String> functions = new ArrayList<>();
-		try{
-			Result rs = session.run("CALL dbms.functions() " +
-					"YIELD name, signature\n" +
-					"RETURN name \n" +
-					"ORDER BY name ASC");
-			while (rs != null && rs.hasNext()) {
-				Record record = rs.next();
-				functions.add(record.get("name").asString());
-			}
-
-			return functions;
-		}catch(Exception e){
-			return Collections.EMPTY_LIST;
+		try {
+			return session.readTransaction(tx -> {
```

Comment:
I can remove that change if you think that's not necessary

Reply:
LGTM

Reply:
What do you think of tracking and changing all `session.run` calls to their read/write transaction equivalents? That might help Aura users willing to use JDBC?

Reply:
@fbiville good idea I created #255